File: data/python/tests/test_request_get.py
import unittest
from function_request_get import RequestHandler


class TestHandleRequest(unittest.TestCase):

    def setUp(self):
        # Create an instance of RequestHandler for each test
        self.handler = RequestHandler()

    def test_success(self):
        result = self.handler.handle_request('https://jsonplaceholder.typicode.com/posts/1')
        self.assertEqual(result, 'success')

    def test_redirect(self):
        result = self.handler.handle_request('https://httpstat.us/301')
        self.assertEqual(result, 'redirect 301')

    def test_error_403(self):
        result = self.handler.handle_request('https://httpstat.us/403')
        self.assertEqual(result, 'error 403')

    def test_error_500(self):
        result = self.handler.handle_request('https://httpstat.us/500')
        self.assertEqual(result, 'error 500')

    def test_timeout(self):
        result = self.handler.handle_request('https://10.255.255.1')
        self.assertEqual(result, 'timeout')


if __name__ == '__main__':
    unittest.main()

File: code/test_suites/cpp/FunctionBuildTest/tests/test_request_get.cpp
#include <gtest/gtest.h>
#include "../src/function_request_get.cpp"

class TestHandleRequest : public ::testing::Test {
protected:
    RequestHandler handler;
};

TEST_F(TestHandleRequest, test_success) {
    std::string result = handler.handle_request("https://jsonplaceholder.typicode.com/posts/1");
    EXPECT_EQ(result, "success");
}

TEST_F(TestHandleRequest, test_redirect) {
    std::string result = handler.handle_request("https://httpstat.us/301");
    EXPECT_EQ(result, "redirect 301");
}

TEST_F(TestHandleRequest, test_error_403) {
    std::string result = handler.handle_request("https://httpstat.us/403");
    EXPECT_EQ(result, "error 403");
}

TEST_F(TestHandleRequest, test_error_500) {
    std::string result = handler.handle_request("https://httpstat.us/500");
    EXPECT_EQ(result, "error 500");
}

TEST_F(TestHandleRequest, test_timeout) {
    std::string result = handler.handle_request("https://10.255.255.1");
    EXPECT_EQ(result, "timeout");
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    int result = RUN_ALL_TESTS();
    return result;
}